src/DynamicNLPModels.jl

```diff
@@ -1173,57 +1288,68 @@ function NLPModels.hess_coord!(
 end
 
 function NLPModels.hess_coord!(
-  qp::LQDynamicModel{T, V, M1, M2, M3},
-  x::AbstractVector{T},
-  vals::AbstractVector{T};
-  obj_weight::Real = one(eltype(x)),
-) where {T, V, M1 <: Matrix, M2 <: Matrix, M3 <: Matrix}
-  NLPModels.increment!(qp, :neval_hess)
-  count = 1
-  for j = 1:(qp.meta.nvar)
-    for i = j:(qp.meta.nvar)
-      vals[count] = obj_weight * qp.data.H[i, j]
-      count += 1
+    qp::DenseLQDynamicModel{T, V, M1, M2, M3},
+    x::AbstractVector{T},
+    vals::AbstractVector{T};
+    obj_weight::Real = one(eltype(x)),
+  ) where {T, V, M1 <: Matrix, M2 <: Matrix, M3 <: Matrix}
+    NLPModels.increment!(qp, :neval_hess)
+    count = 1
+    for j = 1:(qp.meta.nvar)
+      for i = j:(qp.meta.nvar)
+        vals[count] = obj_weight * qp.data.H[i, j]
+        count += 1
+      end
     end
+    return vals
   end
-  return vals
-end
 
 NLPModels.hess_coord!(
-  qp::LQDynamicModel,
+  qp::SparseLQDynamicModel,
+  x::AbstractVector,
+  y::AbstractVector,
+  vals::AbstractVector;
+  obj_weight::Real = one(eltype(x)),
+) = NLPModels.hess_coord!(qp, x, vals, obj_weight = obj_weight)
+
+NLPModels.hess_coord!(
+  qp::DenseLQDynamicModel,
   x::AbstractVector,
   y::AbstractVector,
   vals::AbstractVector;
   obj_weight::Real = one(eltype(x)),
 ) = NLPModels.hess_coord!(qp, x, vals, obj_weight = obj_weight)
 
+
+
 function NLPModels.jac_structure!(
-  qp::LQDynamicModel{T, V, M1, M2, M3},
+  qp::SparseLQDynamicModel{T, V, M1, M2, M3},
   rows::AbstractVector{<:Integer},
   cols::AbstractVector{<:Integer},
 ) where {T, V, M1 <: SparseMatrixCSC, M2 <: SparseMatrixCSC, M3<: AbstractMatrix}
   fill_structure!(qp.data.A, rows, cols)
   return rows, cols
 end
 
+
 function NLPModels.jac_structure!(
-  qp::LQDynamicModel{T, V, M1, M2, M3},
-  rows::AbstractVector{<:Integer},
-  cols::AbstractVector{<:Integer},
-) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
-  count = 1
-  for j = 1:(qp.meta.nvar)
-    for i = 1:(qp.meta.ncon)
-      rows[count] = i
-      cols[count] = j
-      count += 1
+    qp::DenseLQDynamicModel{T, V, M1, M2, M3},
+    rows::AbstractVector{<:Integer},
+    cols::AbstractVector{<:Integer},
+  ) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
+    count = 1
+    for j = 1:(qp.meta.nvar)
```

Comment:
Careful with the indentation: the code should be indented with 4 spaces, consistently (to your defense, I know JSO does not respect that convention, and that's annoying)

Reply:
Note: your editor should be able to reindent the file automatically (`gg=G` in vim for instance)